blockonomics.php

```diff
@@ -406,63 +454,59 @@ public function getContent()
     {
         $output = '';
         if (Tools::isSubmit("testSetup")) {
-            //Save current settings before testing setup
-            Configuration::updateValue(
-                'BLOCKONOMICS_API_KEY',
-                Tools::getValue('BLOCKONOMICS_API_KEY')
-            );
-            $error_str = $this->testSetup();
-            if ($error_str) {
-                $article_url = 'https://blockonomics.freshdesk.com/solution/articles/';
-                $article_url .= '33000215104-troubleshooting-unable-to-generate-new-address';
-                $error_str =
-                    $error_str .
-                    "</br>" .
-                    $this->l('For more information please consult this ') .
-                    "<a target='_blank' href='" .
-                    $article_url. "'>" .
-                    $this->l('troubleshooting article') .
-                    "</a>";
-                $output = $this->displayError($error_str);
+            $this->updateSettings();
+            $api_key = Configuration::get('BLOCKONOMICS_API_KEY');
+            //if there's no API key, give error immediately
+            if (!$api_key) {
```

Comment:
There is duplicate code to check the api_key. This should be refactored into updateSettings() or its own function.

Reply:
Done, I moved it to updateSettings()